--- index.js	
+++ index.js	
@@ -101,18 +101,11 @@
 
 function nodeNC(port, input) {
   const src = nodeNetCatSrc(port, input);
-  if (src.length < 1000) {
-    return spawnSync(process.execPath, ['-e', src], {
-      windowsHide: true,
-      maxBuffer: Infinity,
-    });
-  } else {
     return spawnSync(process.execPath, [], {
       input: src,
       windowsHide: true,
       maxBuffer: Infinity,
     });
-  }
 }
 
 function test(fn, port) {
